src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/ChangeLogIssueKeyExtractor.java

```diff
@@ -80,6 +76,28 @@ public Set<String> extractIssueKeys(
                 .collect(Collectors.toSet());
     }
 
+    private void addFirstNonEmptyChangeSetFromPreviousBuilds(
+            final WorkflowRun workflowRun,
+            final List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets) {
+        if (changeSets.size() == 0) {
```

Comment:
/nit I'd suggest to move this if outside of the function. The name claims it will add the builds, however that's not always right (there's a condition for that which wasn't reflected in the name of the method). Also, I think it'll be better from SRP point of view.